core/consensus/grandpa/impl/grandpa_impl.hpp

```diff
@@ -362,6 +366,13 @@ namespace kagome::consensus::grandpa {
 
     std::vector<WaitingBlock> waiting_blocks_;
 
+    using HistoricalVotesKey = std::pair<AuthoritySetId, RoundNumber>;
+    Lru<HistoricalVotesKey,
+        HistoricalVotesDirty,
+        boost::hash<HistoricalVotesKey>>
+        historical_votes_{5};
+    bool writing_historical_votes_ = false;
```

Comment:
Can it be called from different thread? If true, then it should be atomic

Reply:
No, it's called on grandpa thread